### ISSUE TITLE
torch.Tensor.grad.data attribute is deprecated - update it's usage

### ISSUE THREAD

**Issue by arunppsg** [deepchem/deepchem#3217]
PyTorch is deprecating `torch.Tensor.grad.data`. We are using `data` attribute at some parts of the code, in torch models. The usage should be updated, maybe with `tensor.detach()`?

**Comment by csking101** [comment]
Hi, I'm new to the deepchem community, and am interested to contribute to the organization. I know Python and have worked with basic PyTorch before. Could you assign this issue to me and explain how I can start out with the same?

**Comment by gusty1g** [comment]
Some pointers to start with:
- Figure out why `torch.Tensor.grad` attribute is being used or the purpose of the `grad` attribute in pytorch.
- Find it's occurrence in the code base (you can do a `grep` for `grad.data`), the context or the purpose for which it is used in DeepChem and replace the usage with a suitable solution

Also, see this discussion in pytorch [forum](https://discuss.pytorch.org/t/is-it-good-practice-to-assign-module-weight-grad-data/171254/2)

**Comment by csking101** [comment]
Alright, thank you, I will look into the same and work on the issue.

**Comment by anirudhprabhakaran3** [comment]
Can I have a look at this as well? Would love to research about this a bit.

**Comment by gusty1g** [comment]
> Can I have a look at this as well? Would love to research about this a bit.

Sure, please go ahead.

**Comment by ajain1325** [comment]
Is it open? Can I proceed to work on it.
